### PR TITLE
Add topologies for testing

### PR DESCRIPTION
**Pull request by mauropasse**
Add topologies for testing:
* Single proces pub/sub
* Multi-process pub/sub
* Full topologies
* IPC ON / OFF / Loaned
* Diff messages sizes

### CODE REVIEW

File: irobot_benchmark/CMakeLists.txt [hunk 1]
```diff
@@ -33,12 +33,89 @@ set(TOPOLOGY_FILES
   topology/debug_sierra_nevada_best_effort.json
   topology/debug_mont_blanc_reliable.json
   topology/debug_mont_blanc_best_effort.json
+  topology/debug_white_mountain_best_effort.json
```
Comment (jfinken) [inline irobot_benchmark/CMakeLists.txt:36]: Out of curiosity, what is the purpose/goal of the `debug_*_best_effort.json` topologies?

File: irobot_benchmark/CMakeLists.txt [hunk 2]
```diff
@@ -33,12 +33,89 @@ set(TOPOLOGY_FILES
...
 )
 
 install(FILES
   ${TOPOLOGY_FILES}
   DESTINATION lib/${PROJECT_NAME}/topology)
 
+set(SP topology/tests/single_process)
```
Comment (jfinken) [inline irobot_benchmark/CMakeLists.txt:43]: What do you think about removing the `tests/` subdirectory?  The new topologies are already "identified" via the subdirectories that contain them, like `full_topologies`, `multi_process`, and `single_process`.

File: irobot_benchmark/CMakeLists.txt (hunk 1 above)
Comment (alsora) [reply irobot_benchmark/CMakeLists.txt]: It's a topology with a single node that subscribes to all topics published by the white-mountain.
it simulates a "debug station" connecting to the robot for logging or visualization.

FWIW, these debug topologies could be automatically generated

File: irobot_benchmark/CMakeLists.txt (hunk 2 above)
Comment (jfinken) [reply irobot_benchmark/CMakeLists.txt]: LGTM!  

Another question though, the [white_mountain_fixed_size_debug.json](https://github.com/irobot-ros/ros2-performance/blob/mauro/test-ipc-loaned-3/irobot_benchmark/topology/full_topologies/white_mountain_fixed_size_debug.json) specifies `"qos_reliability":"best_effort"`.  But the analogous [sierra_nevada_fixed_size_debug.json](https://github.com/irobot-ros/ros2-performance/blob/mauro/test-ipc-loaned-3/irobot_benchmark/topology/full_topologies/sierra_nevada_fixed_size_debug.json) does not.  Intentional?

Comment (jfinken) [reply irobot_benchmark/CMakeLists.txt]: I'm seeing a duplicate [debug_white_mountain_best_effort.json](https://github.com/irobot-ros/ros2-performance/blob/mauro/iron-events-executor/irobot_benchmark/topology/debug_white_mountain_best_effort.json) in the top-level `topology` directory.  But it is really (correctly, in my view) beneath the `full_topologies/` directory.  See
- [mauro/iron-events-executor branch](https://github.com/irobot-ros/ros2-performance/tree/mauro/iron-events-executor/irobot_benchmark/topology)
- [mauro/iron-to-galactic branch](https://github.com/irobot-ros/ros2-performance/tree/mauro/iron-to-galactic/irobot_benchmark/topology)

Comment (mauropasse) [reply irobot_benchmark/CMakeLists.txt]: Nice, just removed it.

**Pull request closed without merging**